### PR TITLE
Safarov Nurlan. Laboratory work №4. Option №4.

### PR DESCRIPTION
**Pull request by safarov-nm**
The purpose of this laboratory work is to implement a pass that counts the number of calls to each function in the module.

### CODE REVIEW

File: mlir/lib/Transforms/lab4/safarov_nurlan/countFunctionCallsPass.cpp [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+namespace {
+class FunctionCallCounterPass
+    : public PassWrapper<FunctionCallCounterPass, OperationPass<ModuleOp>> {
+public:
+  StringRef getArgument() const final { return "SafarovNcntFuncCalls"; }
+  void runOnOperation() override {
+    std::vector<LLVM::LLVMFuncOp> functions;
+    std::map<StringRef, int> numberOfCalls;
+
+    getOperation()->walk([&](Operation *opearation) {
```
Comment (m-ly4) [inline mlir/lib/Transforms/lab4/safarov_nurlan/countFunctionCallsPass.cpp]:
```suggestion
    getOperation()->walk([&](Operation *operation) {
```

**Review by m-ly4 — APPROVED** [review]

**Review by aobolensk — APPROVED** [review]

File: mlir/lib/Transforms/lab4/safarov_nurlan/countFunctionCallsPass.cpp (hunk 1 above)
Comment (m-ly4) [reply mlir/lib/Transforms/lab4/safarov_nurlan/countFunctionCallsPass.cpp]: @safarov-nm Could you please fix this typo before we merge it into main branch?

Comment (safarov-nm) [reply mlir/lib/Transforms/lab4/safarov_nurlan/countFunctionCallsPass.cpp]: Okay.

Comment (safarov-nm) [reply mlir/lib/Transforms/lab4/safarov_nurlan/countFunctionCallsPass.cpp]: Fixed it. Thanks!

**Pull request merged**